include/sdf/Param.hh

```diff
@@ -180,13 +184,36 @@ namespace sdf
     /// of the parameter. nullopt otherwise.
     public: std::optional<std::string> GetMaxValueAsString() const;
 
+    /// \brief Set the parameter value from a string.
+    /// \param[in] _value New value for the parameter in string form.
+    /// \param[in] _ignoreParentAttributes Whether to ignore parent element
+    /// attributes when parsing value from string as well as subsequent
+    /// reparses.
+    public: bool SetFromString(const std::string &_value,
+                               bool _ignoreParentAttributes);
+
     /// \brief Set the parameter value from a string.
     /// \param[in] _value New value for the parameter in string form.
     public: bool SetFromString(const std::string &_value);
 
+    /// \brief Get the parent Element of this Param.
+    /// \return Pointer to this Param's parent Element, nullptr if there is no
+    /// parent Element.
+    public: ElementPtr GetParentElement() const;
+
+    /// \brief Set the parent Element of this Param.
+    /// \param[in] _parentElement Pointer to new parent Element. A nullptr can
+    /// provided to remove the current parent Element.
+    public: void SetParentElement(const ElementPtr _parentElement);
+
     /// \brief Reset the parameter to the default value.
     public: void Reset();
 
+    /// \brief Reparse the parameter value, this is usually called after the
+    /// parent element's attributes have been modified.
```

Comment:
Can you add a note here about the behavior after `Set<T>` is called? I'm alluding to this comment https://github.com/ignitionrobotics/sdformat/blob/9ac3504b6889d07c33851f39636ccab9ab1c92b2/test/integration/pose_1_9_sdf.cc#L267-L269

Also, can you perhaps mention when users might want to use `Reparse`?

Reply:
Got it, thanks for pointing that out! https://github.com/ignitionrobotics/sdformat/pull/589/commits/86021804251853244202d81f84cd87080700fed0

Reply:
Looks great! Thanks.